src/lib/icon/icon-registry.ts

```diff
@@ -327,14 +327,24 @@ export class MdIconRegistry {
    */
   private _extractSvgIconFromSet(iconSet: SVGElement, iconName: string): SVGElement {
     const iconNode = iconSet.querySelector('#' + iconName);
+
     if (!iconNode) {
       return null;
     }
+
     // If the icon node is itself an <svg> node, clone and return it directly. If not, set it as
     // the content of a new <svg> node.
-    if (iconNode.tagName.toLowerCase() == 'svg') {
+    if (iconNode.tagName.toLowerCase() === 'svg') {
       return this._setSvgAttributes(iconNode.cloneNode(true) as SVGElement);
     }
+
+    // If the node is a <symbol>, it won't be rendered so we have to convert it into <svg>. Note
```

Comment:
I initially went for the `use` element, but the problem is that, if your page has a `base` tag (which all Angular apps have), Firefox requires you to include the current URL before the hash. E.g. if I was using an icon on `/about`, the `use` element would be `<use href="/about#icon-id"/>`.  The href also would have to be maintained on route changes.